src/header/styles.scss

```diff
@@ -45,7 +45,10 @@
 }
 
 .main {
-  @include styles.text-flex-wrapping;
+  // Like styles.text-flex-wrapping, but without overflow: hidden, to prevent headings from being partially cut off.
+  word-wrap: break-word;
```

Comment:
It's possible that these two rules are also not needed, especially as the child `.title` already has `word-break: break-word` which overrides this behavior. But I prefer to err on the safe side here and might consider further changes later in the scope of #1143.